aws/hll/aws-s3-transfer-manager/src/download.rs

```diff
@@ -34,17 +49,178 @@ impl From<GetObjectInputBuilder> for DownloadRequest {
     }
 }
 
-/// Response type for a single download object request.
-#[derive(Debug)]
-#[non_exhaustive]
-pub struct DownloadResponse {
-    /// Object metadata
-    pub object_meta: ObjectMetadata,
+/// Fluent style builder for [Downloader]
+#[derive(Debug, Clone)]
+pub struct Builder {
+    target_part_size_bytes: u64,
+    checksum_validation_enabled: bool,
+    // TODO(design): should we instead consider an enum here allows for not only explicit but also
+    // an "Auto" mode that allows us to control the concurrency actually used based on overall transfer and part size?
+    concurrency: usize,
+    sdk_config: Option<SdkConfig>,
+}
+
+impl Builder {
+    fn new() -> Self {
+        Self {
+            target_part_size_bytes: 8 * MEBIBYTE,
+            checksum_validation_enabled: true,
+            concurrency: 8,
+            sdk_config: None,
+        }
+    }
+
+    /// Size of parts the object will be downloaded in, in bytes.
+    /// The minimum part size is 5 MiB and any value given less than that will be rounded up.
+    /// Defaults is 8 MiB.
+    pub fn target_part_size(mut self, size_bytes: u64) -> Self {
+        self.target_part_size_bytes = cmp::min(size_bytes, MIN_PART_SIZE);
+        self
+    }
+
+    /// Set whether checksum validation is enabled.
+    /// Default is true.
```

Comment:
When writing doc comments, it's preferred to leave a summary line at the top, then an empty newline, and then whatever else you want to convey. This is b/c of how RustDoc will expose this doc to rust-analyzer and other things.
```suggestion
    /// Set whether checksum validation is enabled.
    ///
    /// Default is true.
```